derive.go

```diff
@@ -70,7 +73,50 @@ func (ek *ExtendedKey) DeriveKey(t *merlin.Transcript) (*ExtendedKey, error) {
 	return ek.key.DeriveKey(t, ek.chaincode)
 }
 
-// DeriveKeySimple derives a subkey identified by byte array i and chain code.
+// HardDeriveMiniSecretKey implements BIP-32 like "hard" derivation of a mini
+// secret from an extended key's secret key
+func (ek *ExtendedKey) HardDeriveMiniSecretKey(i []byte) (*ExtendedKey, error) {
+
+	sk, err := ek.Secret()
+
+	if err != nil {
+		return nil, err
+	}
+
+	msk, chainCode, err := sk.HardDeriveMiniSecretKey(i, ek.chaincode)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return NewExtendedKey(msk, chainCode), nil
```

Comment:
nitpick, could you remove the spaces after the function declaration and before the `if err != nil`s?